jaeger-core/src/main/java/io/jaegertracing/internal/JaegerSpan.java

```diff
@@ -46,7 +46,7 @@
   private List<LogData> logs;
   private boolean finished = false; // to prevent the same span from getting reported multiple times
 
-  JaegerSpan(
+  protected JaegerSpan(
```

Comment:
@isaachier I don't think this will be enough, is it?  I.e. you'd also need an overridable method that actually creates the object.

Reply:
I would prefer if you accompany this with a unit test showing the working subclassing.

Reply:
I added a lot more. The test definitely helped me determine what was necessary.